### PR TITLE
build: Import CocoaAsyncSocket using Carthage

### PR DESCRIPTION
**Pull request by qmfrederik**
This is an alternative to #349, but as I mentioned it gives a compile-time error and I'm not sure how it can be suppressed (other than fixing it upstream)

### CODE REVIEW

**Comment by KazuCocoa** [comment]
oops,  I merged #349

**Comment by mykola-mokhnach** [comment]
@KazuCocoa no problems, simple create another PR to revert it

**Comment by KazuCocoa** [comment]
🆗 
I've created a PR to this PR https://github.com/qmfrederik/WebDriverAgent/pull/2/files
On my local, build succeeded with the change cc @qmfrederik

**Comment by qmfrederik** [comment]
@KazuCocoa Thanks, I merged your PR. It [still fails in CI, though](https://dev.azure.com/quamotion-oss/WebDriverAgent/_build/results?buildId=30&view=logs&j=d983e76e-7f67-5ca5-2d18-82de126f2a8c&t=9426afb3-27b4-517a-4d57-1f232bdd190f)

```
▸ Compiling RoutingHTTPServer.m

❌  /Users/runner/runners/2.170.1/work/1/s/Carthage/Build/iOS/CocoaAsyncSocket.framework/Headers/CocoaAsyncSocket.h:9:1: use of '@import' in framework header is discouraged, including this header requires -fmodules [-Werror,-Watimport-in-framework-header]

@import Foundation;
^



❌  /Users/runner/runners/2.170.1/work/1/s/WebDriverAgentLib/Vendor/RoutingHTTPServer/RoutingHTTPServer.h:20:9: could not build module 'CocoaAsyncSocket'

#import <CocoaAsyncSocket/GCDAsyncSocket.h>
 ~~~~~~~^
```

Are you using an older version of Xcode?

**Comment by KazuCocoa** [comment]
mm, I used Xcode 11.3. Let me try an older one

**Comment by qmfrederik** [comment]
I think the error is specific to Xcode 11.4. See this link: https://developer.apple.com/documentation/xcode_release_notes/xcode_11_4_release_notes?language=objc

**Comment by KazuCocoa** [comment]
https://github.com/qmfrederik/WebDriverAgent/pull/3 could be a workaround...
I was able to build this on Xcode 11.4.1

**Comment by KazuCocoa** [comment]
Note:

For Xcode 11.4+, we can specify `-Wno-atimport-in-framework-header` in Configurations/ProjectSettings.xcconfig to avoid the warning as qmfrederik#3 (since we should avoid the warning in Frameworks/CocoaAsyncSocket.framework)

For less than 11.3, we should not specify the flag since clang raises an error, invalid flag.

If we found a way to handle the flag dynamically, this PR is ready, I believe.

**Comment by KazuCocoa** [comment]
Like https://github.com/qmfrederik/WebDriverAgent/pull/3/files , when I added `-Wno-atimport-in-framework-header` as a flag for the xcodebuild, it was able to build for both 11.4+ and 11.3-.
We need to update documentation about building manually (add below), but it seems reasonable with Carthage so far.

```
#if __has_warning("-Watimport-in-framework-header")	
#pragma clang diagnostic ignored "-Watimport-in-framework-header"	
#endif
```

**Comment by mykola-mokhnach** [comment]
Cannot we simply push a PR to https://github.com/robbiehanson/CocoaAsyncSocket ?

I hope @robbiehanson would be happy to merge it ASAP

**Comment by qmfrederik** [comment]
I've created robbiehanson/CocoaAsyncSocket#723. Let's see how it goes.

**Comment by KazuCocoa** [comment]
Updated with the CocoaAsyncSocket fix
https://github.com/qmfrederik/WebDriverAgent/pull/3

File: WebDriverAgentLib/Routing/FBTCPSocket.h [hunk 1]
```diff
@@ -7,7 +7,11 @@
  * of patent rights can be found in the PATENTS file in the same directory.
  */
 
-#import "GCDAsyncSocket.h"
+#if __has_warning("-Watimport-in-framework-header")
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Routing/FBTCPSocket.h]: I would say this is not necessary anymore after GCDAsyncSocket bump

**Review by KazuCocoa — APPROVED** [review]
yey

**Pull request merged**